x/gamm/pool-models/concentrated-liquidity/pool.go

```diff
@@ -10,6 +12,67 @@ var (
 	_ types.PoolI = &Pool{}
 )
 
+func NewConcentratedLiquidityPool(poolId uint64) (Pool, error) {
+	poolAddr := types.NewPoolAddress(poolId)
+
+	// pool thats created up to ensuring the assets and params are valid.
+	// We assume that FuturePoolGovernor is valid.
+	pool := &Pool{
+		Address: poolAddr.String(),
+		Id:      poolId,
+	}
+
+	return *pool, nil
```

Comment:
I'm still down to remove the error since we are likely to not need it, forget and leave it